File: src/main/java/io/github/tanguygab/conditionalactions/actions/types/items/GiveItemAction.java
package io.github.tanguygab.conditionalactions.actions.types.items;

import io.github.tanguygab.conditionalactions.actions.types.Action;
import org.bukkit.Material;
import org.bukkit.OfflinePlayer;
import org.bukkit.entity.Player;
import org.bukkit.inventory.ItemStack;
import org.bukkit.inventory.meta.ItemMeta;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class GiveItemAction extends Action {

    public GiveItemAction() {
        super("(i?)give-item:( )?");
    }

    @Override
    public String getSuggestion() {
        return "give-item: <material> <amount> <name\\nlore>";
    }

    @Override
    public void execute(OfflinePlayer player,String match) {
        if (!(player instanceof Player p)) return;
        String[] args = match.split(" ");

        String material = args[0].toUpperCase();
        Material mat = Material.getMaterial(material);
        if (mat == null) return;
        ItemStack item = new ItemStack(mat);

        ItemMeta meta = item.getItemMeta();
        assert meta != null;

        if (args.length > 1) item.setAmount(parseInt(parsePlaceholders(p,args[1]),1));
        if (args.length > 2) {
            args = Arrays.copyOfRange(args,2,args.length);
            args = String.join(" ",args).split("\\n");
            meta.setDisplayName(parsePlaceholders(p,args[0]));

            if (args.length > 1) {
                List<String> lore = new ArrayList<>();
                for (String line : Arrays.copyOfRange(args,1,args.length))
                    lore.add(parsePlaceholders(p, line));
                meta.setLore(lore);
            }
        }
        item.setItemMeta(meta);
        p.getInventory().addItem(item);
    }
}

File: src/main/kotlin/io/github/tanguygab/conditionalactions/actions/types/items/GiveItemAction.kt
package io.github.tanguygab.conditionalactions.actions.types.items

import io.github.tanguygab.conditionalactions.actions.Action
import net.kyori.adventure.text.Component
import org.bukkit.Material
import org.bukkit.OfflinePlayer
import org.bukkit.entity.Player
import org.bukkit.inventory.ItemStack


class GiveItemAction : Action("^(i?)give-item:( )?") {
    override fun getSuggestion() = "give-item: <material> <amount> <name\\nlore>"

    override fun execute(player: OfflinePlayer?, match: String) {
        if (player !is Player) return
        var args = match.split(" ")

        val material = args[0].uppercase()
        val mat = Material.getMaterial(material) ?: return
        val item = ItemStack(mat)


        if (args.size > 1) item.amount = parseInt(parsePlaceholders(player, args[1]), 1)
        val meta = item.itemMeta
        if (args.size > 2 && meta != null) {
            args = args.subList(2, args.size)
            args = args.joinToString(" ").split("\\n")
            meta.displayName(mm.deserialize(parsePlaceholders(player, args[0])))

            if (args.size > 1) {
                val lore = mutableListOf<Component>()
                for (line in args.subList(1, args.size))
                    lore.add(mm.deserialize(parsePlaceholders(player,line)))
                meta.lore(lore)
                item.itemMeta = meta
            }
        }
        player.inventory.addItem(item)
    }
}
